mnemosyne/core/beam.py

```
@@ -482,30 +501,59 @@ def _get_connection(db_path: Path = None) -> sqlite3.Connection:
                 sqlite_vec.load(conn)
             except Exception:
                 pass  # Some environments don't support load_extension
-        _thread_local.conn = conn
-        _thread_local.db_path = str(path)
-    return _thread_local.conn
+        connections[key] = conn
+    return conn
+
+
+def _acquire_beam_connection(db_path: Path = None) -> sqlite3.Connection:
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)